src/olympia/reviewers/management/commands/notify_about_auto_approve_delay.py

```diff
@@ -17,21 +18,36 @@ def fetch_versions_waiting_for_approval_for_too_long(self):
         """
         Return versions that are auto-approvable but have waited long enough
         that we need to notify their developers about the delay.
+
+        Versions are grouped by add-on, only the latest auto-approvable version
+        for each add-on is considered ; if a version has been created recently
+        enough for a given add-on and not auto-approved, none of its other
+        versions will be returned.
         """
-        waiting_period = datetime.now() - timedelta(
-            hours=self.WAITING_PERIOD_HOURS)
         exclude_kwargs = {
             'addon__addonreviewerflags__notified_about_auto_approval_delay':
                 True
         }
-        return (
+        qs = (
+            # Base queryset with auto-approvable versions. Default ordering
+            # is reset to make the GROUP BY work.
             Version.objects
                    .auto_approvable()
-                   .filter(created__lt=waiting_period)
                    .exclude(**exclude_kwargs)
-                   .order_by('created')
-                   .distinct()
```

Comment:
The `distinct()` is no longer needed now that we're grouping by add-on in the subquery and then filtering by pk.